src/modules/simulator/accelsim/accelsim.cpp

```diff
@@ -318,65 +315,45 @@ ACCELSIM::~ACCELSIM()
 int
 ACCELSIM::init()
 {
-	int ret = -1;
-
-	struct mag_report mrp = {};
-	sensor_accel_s arp = {};
-
 	/* do SIM init first */
 	if (VirtDevObj::init() != 0) {
 		PX4_WARN("SIM init failed");
-		goto out;
+		return PX4_ERROR;
 	}
 
 	/* allocate basic report buffers */
 	_accel_reports = new ringbuffer::RingBuffer(2, sizeof(sensor_accel_s));
 
 	if (_accel_reports == nullptr) {
-		goto out;
+		return PX4_ERROR;
 	}
 
-	_mag_reports = new ringbuffer::RingBuffer(2, sizeof(mag_report));
+	_mag_reports = new ringbuffer::RingBuffer(2, sizeof(sensor_mag_s));
 
 	if (_mag_reports == nullptr) {
-		goto out;
+		return PX4_ERROR;
 	}
 
 	/* do init for the mag device node */
-	ret = _mag->init();
+	int ret = _mag->init();
 
 	if (ret != OK) {
 		PX4_WARN("MAG init failed");
-		goto out;
+		return PX4_ERROR;
 	}
 
 	/* fill report structures */
 	_measure();
 
-	/* advertise sensor topic, measure manually to initialize valid report */
-	_mag_reports->get(&mrp);
-
-	/* measurement will have generated a report, publish */
-	_mag->_mag_topic = orb_advertise_multi(ORB_ID(sensor_mag), &mrp,
-					       &_mag->_mag_orb_class_instance, ORB_PRIO_LOW);
-
-	if (_mag->_mag_topic == nullptr) {
-		PX4_WARN("ADVERT ERR");
-	}
+	mag_report mag_report = {};
 
-	/* advertise sensor topic, measure manually to initialize valid report */
-	_accel_reports->get(&arp);
+	/* advertise mag sensor topic, measure manually to initialize valid report */
+	_mag_reports->get(&mag_report);
 
 	/* measurement will have generated a report, publish */
-	_accel_topic = orb_advertise_multi(ORB_ID(sensor_accel), &arp,
-					   &_accel_orb_class_instance, ORB_PRIO_DEFAULT);
-
-	if (_accel_topic == nullptr) {
-		PX4_WARN("ADVERT ERR");
-	}
+	orb_publish_auto(ORB_ID(sensor_mag), &_mag->_mag_topic, &mag_report, &_mag->_mag_orb_class_instance, ORB_PRIO_LOW);
```

Comment:
I was not clear enough: I was thinking to remove the `orb_publish_auto` from here completely. If there is no ordering issue you don't need it here.
If you want to have a separate advertise in the init phase, it's better to use `orb_advertise_multi`, since you already know it's not yet initialized.

Reply:
Done. Thanks!  It is unclear to me if the`_mag_reports->get(&mag_report);` call is still required as it looks like the `_mag_reports->force(&mag_report);` call in `mag_measure()` will accomplish what is taking place there. I might be overlooking something so I have left the other lines in place.  Let me know if your intent was to deprecate those three lines of code, (which it looks like we can do):
```
	sensor_mag_s mag_report = {};

	/* advertise mag sensor topic, measure manually to initialize valid report */
	_mag_reports->get(&mag_report);

	/* measurement will have generated a report, publish */
	orb_publish_auto(ORB_ID(sensor_mag), &_mag->_mag_topic, &mag_report, &_mag->_mag_orb_class_instance, ORB_PRIO_LOW);
```